sound/soc/sof/topology.c

```diff
@@ -1057,7 +1057,7 @@ static int sof_control_load_bytes(struct snd_soc_component *scomp,
 	/* init the get/put bytes data */
 	scontrol->size = sizeof(struct sof_ipc_ctrl_data) +
 		le32_to_cpu(control->priv.size);
-	scontrol->control_data = kzalloc(max_size, GFP_KERNEL);
+	scontrol->control_data = kzalloc(scontrol->size, GFP_KERNEL);
```

Comment:
@ranj063  Thanks, I think I got it. So the problem in my case is that the topology sets max_size to a too smaller value. In my case priv size is 40, max_size is 48 and sizeof(sof_ipc_ctrl_data) is 92. 
Adjusting the `kzalloc()` size here wouldn't help as size in `snd_sof_bytes_get/put()` would be higher than `be_max` so those would fail every time. I guess the solution for this would be to simply issue a topology load error if `max_size < scontrol->size`. Is this correct?

Reply:
@dragosht yes, that correct. We already have a check  but its not quite right here:
https://github.com/thesofproject/linux/blob/4e76b4a1623eb90f42b45147420d03ebbe48f9a4/sound/soc/sof/topology.c#L1051

It needs to be changed to 	if (scontrol->size > max_size)  after scontrol_size is set